test/integration_test/kubevirt_hyperconv_test.go

```diff
@@ -0,0 +1,499 @@
+		}
+	}
+	if transitionTime.IsZero() {
+		return "", "", time.Time{}, fmt.Errorf(
+			"failed to determine when VMI %s/%s transitioned to phase %s", vmiNamespace, vmiName, vmi.Phase)
+	}
+	return vmi.UID, vmi.Phase, transitionTime, nil
+}
+
+// Get mount type (nfs or bind) of the root disk volume. This function assumes that the volume name inside
+// the pod yaml is "rootdisk". e.g.
+//
+//	  volumes:
+//	  - name: rootdisk
+//	    persistentVolumeClaim:
+//		  claimName: fedora-communist-toucan
+func getVMRootDiskMountType(pod *corev1.Pod, apiVol *api.Volume) (string, error) {
+	podNamespacedName := pod.Namespace + "/" + pod.Name
+	logrus.Infof("checking the rootdisk mount type in pod %s", podNamespacedName)
+
+	// Sample output if the volume is bind-mounted:
+	// $ kubectl exec -it virt-launcher-fedora-communist-toucan-jfw7n -- mount
+	// ...
+	// /dev/pxd/pxd365793461222635857 on /run/kubevirt-private/vmi-disks/rootdisk type ext4 (rw,relatime,seclabel,discard)
+	// ...
+	bindMountRE := regexp.MustCompile(fmt.Sprintf("/dev/pxd/pxd%s on .*rootdisk type (ext4|xfs)", apiVol.Id))
+
+	// Sample output if the volume is nfs-mounted:
+	// $ kubectl exec -it virt-launcher-fedora-communist-toucan-bqcrp -- mount
+	// ...
+	// 172.30.194.11:/var/lib/osd/pxns/365793461222635857 on /run/kubevirt-private/vmi-disks/rootdisk type nfs (...)
+	// ...
+	nfsMountRE := regexp.MustCompile(fmt.Sprintf(":/var/lib/osd/pxns/%s on .*rootdisk type nfs", apiVol.Id))
+
+	cmd := []string{"mount"}
+	output, err := core.Instance().RunCommandInPod(cmd, pod.Name, "", pod.Namespace)
+	if err != nil {
+		return "", fmt.Errorf("failed to run command %v inside the pod %s", cmd, podNamespacedName)
+	}
+	var foundBindMount, foundNFSMount bool
+	for _, line := range strings.Split(output, "\n") {
+		if bindMountRE.MatchString(line) {
+			if foundBindMount || foundNFSMount {
+				return "", fmt.Errorf("multiple rootdisk mounts found: %s", output)
+			}
+			foundBindMount = true
+			logrus.Infof("found root disk bind mounted for VM pod %s: %s", podNamespacedName, line)
+		}
+
+		if nfsMountRE.MatchString(line) {
+			if foundBindMount || foundNFSMount {
+				return "", fmt.Errorf("multiple rootdisk mounts found: %s", output)
+			}
+			foundNFSMount = true
+			logrus.Infof("found root disk nfs mounted for VM pod %s: %s", podNamespacedName, line)
+		}
+	}
+	if !foundBindMount && !foundNFSMount {
+		return "", fmt.Errorf("no root disk mount in pod %s: %s", podNamespacedName, output)
+	}
+	if foundBindMount {
+		return mountTypeBind, nil
+	}
+	return mountTypeNFS, nil
+}
+
+func verifyVMStayedUp(t *testing.T, testState *kubevirtTestState) {
+	vmiUIDAfter, vmiPhaseAfter, transitionTimeAfter, err := getVMIDetails(testState.vmPod.Namespace, testState.vmiName)
+	require.NoError(t, err)
+	require.Equal(t, "Running", vmiPhaseAfter)
+	require.Equal(t, testState.vmiUID, vmiUIDAfter)
+	require.Equal(t, testState.vmiPhaseTransitionTime, transitionTimeAfter)
```

Comment:
Just curious whether this is expected to match every time. Is it not a function of the system load etc?

Reply:
I verified that if a VM is stopped and started again, a new VMI object gets created with the same name (i.e. the UID will change). We are using that fact here to ensure that VM did not stop during our test. Let me know if that answers your question. I did not understand reference to system load.